<a href="https://colab.research.google.com/github/aid4mh/QPrism/blob/main/tests/Sensor/RLC_testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#This notebook tests the correctness of RLC using the generated data from [here](https://github.com/aid4mh/QPrism/tree/main/data/sensor/generated_testing_data).

In [1]:
! python3 -m pip install --upgrade pip

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 2.0 MB 30.9 MB/s 
  Attempting uninstall: pip
    Found existing installation: pip 21.1.3
    Uninstalling pip-21.1.3:
      Successfully uninstalled pip-21.1.3


#Install the dependencies and QPrism

In [ ]:
! pip install -r https://raw.githubusercontent.com/aid4mh/QPrism/main/requirements.txt

In [34]:
! pip install --no-deps QPrism

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [83]:
from QPrism.Sensor.DQM import DQM_multiple_record
from QPrism.Sensor.load_data import *

#For two records both have 3 rows of data, RLC should be 1..

In [80]:
import pandas as pd

df_list = []
df_list.append(pd.read_csv("https://raw.githubusercontent.com/aid4mh/QPrism/main/data/sensor/generated_testing_data/RLC_3_rows_1.csv"))
df_list.append(pd.read_csv("https://raw.githubusercontent.com/aid4mh/QPrism/main/data/sensor/generated_testing_data/RLC_3_rows_1.csv"))
for df in df_list:
    print(df)

   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
2    0.03 -0.340 -0.918  1.160
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
2    0.03 -0.340 -0.918  1.160


In [85]:
testing_RLC_DQM = DQM_multiple_record()
testing_RLC_DQM.set_input_data(df_list)
testing_RLC_DQM.compute_avg_DQM()

Start computing the DQM... This may take a long time if the dataset is large
The total time for computing the DQM is: 0.057708740234375 seconds.


#The RLC score is 1 as expected

In [86]:
testing_RLC_DQM.get_RLC()

'1.0'

#Now we adding one more record that has 2 rows of data to the list. Now the record lengths are not the same across records, RLC shoud be lower than 1.

In [88]:
df_list.append(pd.read_csv("https://raw.githubusercontent.com/aid4mh/QPrism/main/data/sensor/generated_testing_data/RLC_2_rows.csv"))
for df in df_list:
    print(df)

   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
2    0.03 -0.340 -0.918  1.160
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
2    0.03 -0.340 -0.918  1.160
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031


In [89]:
testing_RLC_DQM.set_input_data(df_list)
testing_RLC_DQM.compute_avg_DQM()

Start computing the DQM... This may take a long time if the dataset is large
The total time for computing the DQM is: 0.08531379699707031 seconds.


# The RLC score is less than 1 as expected

In [90]:
testing_RLC_DQM.get_RLC()

'0.911841113301548'

#Now we again adding one more record that has 2 rows of data to the list. Now RLC shoud be lower than 0.91184 that computed from provious steps since the it is more inconsistent, but should not be differed by a large value, as [3, 3, 2, 2] are not significantly inconsistent.

In [91]:
df_list.append(pd.read_csv("https://raw.githubusercontent.com/aid4mh/QPrism/main/data/sensor/generated_testing_data/RLC_2_rows_2.csv"))
for df in df_list:
    print(df)

   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
2    0.03 -0.340 -0.918  1.160
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
2    0.03 -0.340 -0.918  1.160
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031


In [92]:
testing_RLC_DQM.set_input_data(df_list)
testing_RLC_DQM.compute_avg_DQM()

Start computing the DQM... This may take a long time if the dataset is large
The total time for computing the DQM is: 0.2597074508666992 seconds.


#The RLC score is lower than 0.91184, but not significantly lower. Which is expected.

In [93]:
testing_RLC_DQM.get_RLC()

'0.9003320053750441'

#Again adding one more record that has 20 rows of data to the list. Now RLC shoud be significantly lower than 0.90033 that computed from provious steps, as 20 addes 'a lot of inconsistency'.

In [94]:
df_list.append(pd.read_csv("https://raw.githubusercontent.com/aid4mh/QPrism/main/data/sensor/generated_testing_data/RLC_20_rows.csv"))
for df in df_list:
    print(df)

   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
2    0.03 -0.340 -0.918  1.160
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
2    0.03 -0.340 -0.918  1.160
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
   time_s   lw_x   lw_y   lw_z
0    0.01 -0.004  0.945 -0.461
1    0.02 -0.406 -0.379  1.031
    time_s   lw_x   lw_y   lw_z
0     0.01 -0.004  0.945 -0.461
1     0.02 -0.406 -0.379  1.031
2     0.03 -0.004  0.945 -0.461
3     0.04 -0.406 -0.379  1.031
4     0.05 -0.004  0.945 -0.461
5     0.06 -0.406 -0.379  1.031
6     0.07 -0.004  0.945 -0.461
7     0.08 -0.406 -0.379  1.031
8     0.09 -0.004  0.945 -0.461
9     0.10 -0.406 -0.379  1.031
10    0.11 -0.004  0.945 -0.461
11    0.12 -0.406 -0.379  1.031
12    0.13 -0.004  0.945 -0.461
13    0.14 -0.406 -0.379  1.031
14    0.15 -0.004  0.945 -0.461
15    0.16 -0.406 -0.379  1.031
16    0.17 -0.004  0.9

In [96]:
testing_RLC_DQM.set_input_data(df_list)
testing_RLC_DQM.compute_avg_DQM()

Start computing the DQM... This may take a long time if the dataset is large
The total time for computing the DQM is: 0.14041590690612793 seconds.


#The RLC score is significantly lower than 0.91184, which is expected.

In [97]:
testing_RLC_DQM.get_RLC()

'0.4740552264497351'